cdap-app-fabric/src/main/java/io/cdap/cdap/internal/profile/ProfileService.java

```diff
@@ -142,10 +148,13 @@ public Profile getProfile(ProfileId profileId, Map<String, String> overrides) th
    *
    * @param profileId the id of the profile to save
    * @param profile the information of the profile
-   * @throws MethodNotAllowedException if trying to update the Native profile
+   * @throws MethodNotAllowedException if trying to update the Native profile or creating new profile when disallowed.
    */
   public void saveProfile(ProfileId profileId, Profile profile) throws MethodNotAllowedException {
     TransactionRunners.run(transactionRunner, context -> {
+      if (!cConf.getBoolean(Constants.Profile.CREATION_ALLOWED, true)) {
+        throw new MethodNotAllowedException(String.format("Creating new profile is disabled"));
```

Comment:
No need to use `String.format`. Also, the message is better as "Compute profile creation is not allowed".

Reply:
Done. Fixed